### PR TITLE
#184: Upgrade dependencies

### LINKED ISSUE #184: Update test dependencies of extension-manager-integration-test-java to fix vulnerabilities
CVE-2024-47535: io.netty:netty-common:jar:4.1.108.Final:test
see https://github.com/exasol/oracle-virtual-schema/issues/50

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #184

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ linters:
     - gofumpt # "gofmt" is OK for us, no need for stricter rules (https://github.com/mvdan/gofumpt)
     - gci # No need to explicitly format imports using GCI (https://github.com/daixiang0/gci)
     - depguard # No need to restrict imported packages (https://github.com/OpenPeeDeeP/depguard)
-    - goerr113 # Using static errors is not necessary (https://github.com/Djarvur/go-err113)
+    - err113 # Using static errors is not necessary (https://github.com/Djarvur/go-err113)
```
Comment (kaklakariada) [inline .golangci.yml:22]: Note: upgrade of linter required changes in this config file and added new warnings. I fixed them all.

File: pkg/backend/ExasolSqlClient.go [hunk 2]
```diff
@@ -79,7 +79,7 @@ func closeRows(rows *sql.Rows) error {
 	return nil
 }
 
-func (c exasolSqlClient) extractResult(rows *sql.Rows) (*QueryResult, error) {
+func (c *exasolSqlClient) extractResult(rows *sql.Rows) (*QueryResult, error) {
```
Comment (kaklakariada) [inline pkg/backend/ExasolSqlClient.go:82]: Linter finding: consistently use pointer receiver.

File: pkg/integrationTesting/scriptFixtures.go [hunk 3]
```diff
@@ -88,6 +88,7 @@ func (f ScriptFixture) GetMetaDataSchemaName() string {
 }
 
 func (f ScriptFixture) Cleanup(t *testing.T) {
+	t.Helper()
```
Comment (kaklakariada) [inline pkg/integrationTesting/scriptFixtures.go:91]: Linter finding: call this method first in all helper methods.

File: pkg/backend/ExasolSqlClient_i_test.go [hunk 4]
```diff
@@ -70,13 +70,13 @@ func (suite *ExasolSqlClientITestSuite) TestQueryFails() {
 func (suite *ExasolSqlClientITestSuite) TestQuerySucceeds() {
 	result, err := suite.client.Query("select 1 as col")
 	suite.Require().NoError(err)
-	suite.Equal(&backend.QueryResult{Columns: []backend.Column{{Name: "COL", TypeName: "DECIMAL"}}, Rows: []backend.Row{{(float64(1))}}}, result)
+	suite.Equal(&backend.QueryResult{Columns: []backend.Column{{Name: "COL", TypeName: "DECIMAL"}}, Rows: []backend.Row{{(int64(1))}}}, result)
```
Comment (kaklakariada) [inline pkg/backend/ExasolSqlClient_i_test.go:73]: Seems to be caused by updated JSON in WebSocket protocol of Exasol 8.32.0

File: pom.xml [hunk 5]
```diff
@@ -9,8 +9,9 @@
     <packaging>pom</packaging>
     <url>https://github.com/exasol/extension-manager/</url>
     <properties>
-        <revision>0.5.12</revision>
+        <revision>0.5.13</revision>
         <junit.version>5.10.2</junit.version>
+        <java.version>11</java.version>
```
Comment (kaklakariada) [inline pom.xml:14]: Explicitly set Java version after upgrading PK

**Review by ckunki — DISMISSED** [review]

**Review by ckunki — APPROVED** [review]

**Pull request merged**